clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -183,25 +183,28 @@ public void poll(long timeout) {
 
     private void poll(long timeout, long now) {
         // send all the requests we can send now
-        pollUnsentRequests(now);
-        now = time.milliseconds();
-        
+        trySend(now);
+
         // ensure we don't poll any longer than the deadline for
         // the next scheduled task
         timeout = Math.min(timeout, delayedTasks.nextTimeout(now));
         clientPoll(timeout, now);
+        now = time.milliseconds();
+
+        // handle any disconnects by failing the active requests. note that disconects must
+        // be checked immediately following poll since any subsequent call to client.ready()
+        // will reset the disconnect status
+        checkDisconnects(now);
 
         // execute scheduled tasks
-        now = time.milliseconds();
         delayedTasks.poll(now);
 
         // try again to send requests since buffer space may have been
         // cleared or a connect finished in the poll
-        pollUnsentRequests(now);
+        trySend(now);
```

Comment:
I was thinking of cases where multiple requests needed to be pushed to the same broker. Most of the time, the socket will happily accept the requests, but with the current implementation, requests are only pushed when the socket is selected for writing in Selector (I recall the netty guy advises against doing this, by the way), so you have to do the ready/poll dance. In the end, I don't think we have any cases in the consumer where we actually need to pipeline a bunch of requests, so I wanted ConsumerNetworkClient.poll() to have only one actual invocation of NetworkClient.poll().